### PR TITLE
Revert "fix: preserve mtime when zipping with the node zipper (#539)"

### PR DESCRIPTION
**Pull request by bnegrao**
This PR reverts commit 162d9bc7f0e87d63cd93adb0f4756d01769da79d, which introduced non-deterministic behavior in the bundling process, leading to unnecessary re-deployments even when no code changes occurred. This change addresses issue [#557](https://github.com/floydspace/serverless-esbuild/issues/557).

### CODE REVIEW

**Comment by floydspace** [comment]
hi @bnegrao , thanks, I think it's fair to revert it, @iainlane we might need to have a better solution for this, like having configurtation option for zipper, for now I just let it revert

**Review by floydspace — APPROVED** [review]

**Comment by iainlane** [comment]
Any idea what the problem is? Is it to do with the target system or something?

**Comment by iainlane** [comment]
A testcase to reproduce would be nice; the one which is being reverted here is asserting that the results _are_ deterministic.

**Comment by iainlane** [comment]
Actually, I suspect what is happening is that generated files are updated on a re-package. Isn't that what is expected? The zip _should_ be different if the input files have been updated (touched).

So - I think reverting this was the wrong move. The old behaviour wasn't really correct, and the new one was. If you're repackaging and that involves updating the generated code, you should indeed expect a different checksum on your zip.

In other words:

> The zip files should be bundled in a "deterministic" way, that means, bundling the same code should always produce the exact same bundle.

Disagree that we are seeing "nondeterminism".

**Comment by iainlane** [comment]
I suspect if we took a look at the `ls -lR` of the zips we'd see the (generated) files have different mtimes.

Unless I understand [this](https://github.com/bnegrao/serverless-esbuild/blob/17cc8e6365f82accc541fcfc59cd2096edd862c2/src/utils.ts#L109
) wrong (could be), we're taking the mtime of the source file itself - the file which is being added to the zip a few lines later. (On the next line we skip directories)

**Comment by floydspace** [comment]
> Isn't that what is expected?
this is what I expect: no matter what changes are in the source code, if bundling produces the same bundle from content point of view, it should not trigger cloudformation changeset

**Comment by iainlane** [comment]
Well if that's the position then you disagree with my initial change as far as I can see. When are people doing this kind of no-change repackage?

If you ask me I'd state it as this: if you run a no-op rebuild (why?) then you should expect to get a redeploy, because your files did change - even if just in modification time.

Which is a conflicting position to yours, and unless you find that convincing, as the maintainer your view should be the one to prevail.

It's no problem. I'll go and look for another way to deploy in which I can have working bundled static assets. :-)
